### ISSUE TITLE
Question on log_type 

### ISSUE THREAD

**Issue by kapinyajudit** [yokawasa/logstash-output-azure_loganalytics#13]
@yokawasa is it possible to use a variable as log_type?

e.g.
```
if [smth] {
      azure_loganalytics {
        customer_id => "..."
        shared_key => "..."
        log_type => "%{[log_type_name]}"
        key_names  => ['timestamp', 'message', 'er']
      }
}
```

I always get 
```
exception=>#<ArgumentError: log_type must be only alpha characters>
```

**Comment by yokawasa** [comment]
@kapinyajudit 
I haven't try but i think it's possible. please check this and try. Anyway, very sorry for the late response

- https://www.elastic.co/guide/en/logstash/current/field-references-deepdive.html#field-references-deepdive

**Comment by cmaenner** [comment]
Same issue on my end. I am trying to dynamically write to the `Custom Logs Table` associated with the Azure Event Hub topic:

```
# Adds meta data to the event.
# [@metadata][azure_event_hubs][name] - the name of the event host
# [@metadata][azure_event_hubs][consumer_group] - the consumer group that consumed this event
# [@metadata][azure_event_hubs][processor_host] - the unique identifier that identifies which host processed this event. Note - there can be multiple processor hosts on a single instance of Logstash.
# [@metadata][azure_event_hubs][partition] - the partition from which event came from
# [@metadata][azure_event_hubs][offset] - the event hub offset for this event
# [@metadata][azure_event_hubs][sequence] - the event hub sequence for this event
# [@metadata][azure_event_hubs][timestamp] - the enqueued time of the event
# [@metadata][azure_event_hubs][event_size] - the size of the event

input {
  azure_event_hubs {
    threads => 8
    decorate_events => true
    event_hub_connections => [""]
    storage_connection => ""
  }
}

filter {
  json {
    source => "message"
    target => "raw"
    remove_field => ["message"]
  }
  split {
    field => "raw"
  }
  mutate {
    add_field => {
      "id" => "%{[raw][id]}"
      "date_create" => "%{[raw][date_create]}"
      "action" => "%{[raw][action]}"
      "actor_user_id" => "%{[raw][actor][user][id]}"
      "actor_user_name" => "%{[raw][actor][user][name]}"
      "actor_user_email" => "%{[raw][actor][user][email]}"
      "actor_type" => "%{[raw][actor][type]}"
      "context_location_type" => "%{[raw][context][location][type]}"
      "context_location_id" => "%{[raw][context][location][id]}"
      "context_location_name" => "%{[raw][context][location][name]}"
      "context_location_domain" => "%{[raw][context][location][domain]}"
      "context_ua" => "%{[raw][context][ua]}"
      "context_ip_address" => "%{[raw][context][ip_address]}"
      "entity_type" => "%{[raw][entity][type]}"
      "entity_file_id" => "%{[raw][entity][file][id]}"
      "entity_file_name" => "%{[raw][entity][file][name]}"
      "entity_file_filetype" => "%{[raw][entity][file][filetype]}"
      "entity_file_title" => "%{[raw][entity][file][title]}"
      "event_hub_timestamp" => "%{[@metadata][azure_event_hubs][timestamp]}"
    }
    remove_field => ["@version"]
  }
}

output {
  azure_loganalytics {
    customer_id => ""
    shared_key => ""
    log_type => "%{[@metadata][azure_event_hubs][name]}"
    key_names => ["id", "date_create", "action", "actor_user_id", "actor_user_name", "actor_user_email", "actor_type", "context_location_type", "context_location_id", "context_location_name", "context_location_domain", "context_ua", "context_ip_address", "entity_type", "entity_file_id", "entity_file_name", "entity_file_filetype", "entity_file_title", "event_hub_timestamp"]
    flush_items => 50
    flush_interval_time => 5
  }
}
```

**Comment by kapinyajudit** [comment]
It seems it does not support the `%{...}` substitution. In my local version I made a little change which allowed me to use dynamic values (ofc I had to get rid of the check for alpha characters - but in my case it is always alpha):
`@log_type = event.sprintf(log_type)`
<img width="425" alt="Screen Shot 2020-06-04 at 22 39 12" src="https://user-images.githubusercontent.com/5430276/83807984-3ca51f80-a6b4-11ea-9961-ffa37ad1eaf0.png">

**Comment by cmaenner** [comment]
Cool, thanks @kapinyajudit I will try to change the source locally and see if it fixes the issue. Appreciate your help.

**Comment by yokawasa** [comment]
@kapinyajudit 
thank you so much for the issue. I thought it very useful feature, i mean sprintf syntax. So I've just published `logstash-output-azure_loganalytics-0.5.0` where
* Support sprintf syntax like `%{my_log_type}` for `log_type` config param - [Issue #13](https://github.com/yokawasa/logstash-output-azure_loganalytics/issues/13)

Here is new description of `log_type` param (See also [README.md](https://github.com/yokawasa/logstash-output-azure_loganalytics/blob/master/README.md))
> The name of the event type that is being submitted to Log Analytics. It must only contain alpha numeric and _, and not exceed 100 chars. sprintf syntax like %{my_log_type} is supported.


relevant links
- https://github.com/yokawasa/logstash-output-azure_loganalytics/releases/tag/v0.5.0
- https://rubygems.org/gems/logstash-output-azure_loganalytics/versions/0.5.0

Thanks again.  I'm closing the issue
